WordPress/gutenberg
# Writing Flow: Collapse range in horizontal edge check by selection direction

Fixes #5095

This pull request seeks to resolve an issue where it is not possible to unexpand a selection from the beginning of a paragraph block. I'm not entirely clear the intent with collapsing range in the `isHorizontalEdge` function, though it would seem that it should be based on the direction of the current selection itself, not necessarily the direction being considered as the edge.

__Testing instructions:__

Verify that various shift-selection within and across blocks works as expected, particularly the previously failing behavior:

1. Insert a new paragraph block with some text
2. Make a selection of the text from the beginning of the block
3. Press Shift+ArrowLeft to unexpand the selection
4. Note that the selection becomes unexpanded as expected
